File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/api/ApiManager.java
package com.androidcollider.easyfin.common.managers.api;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * @author Ihor Bilous
 */

public class ApiManager {

    private static final String BASE_URL = "http://api.minfin.com.ua";

    private Retrofit mRetrofit;


    ApiManager() {
        Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create());

        mRetrofit = retrofitBuilder
                .client(new OkHttpClient.Builder().build())
                .build();
    }

    public Retrofit getRetrofit() {
        return mRetrofit;
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/api/ApiManager.kt
package com.androidcollider.easyfin.common.managers.api

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * @author Ihor Bilous
 */
class ApiManager internal constructor() {
    val retrofit: Retrofit

    init {
        val logging = HttpLoggingInterceptor()
        logging.setLevel(HttpLoggingInterceptor.Level.BODY)
        val httpClient = OkHttpClient.Builder()
        httpClient.addInterceptor(logging)
        httpClient.cache(null)
        val retrofitBuilder = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
        retrofit = retrofitBuilder
            .client(httpClient.build())
            .build()
    }

    companion object {
        private const val BASE_URL = "https://bank.gov.ua/NBUStatService/v1/statdirectory/"
    }
}